datafusion/functions/src/string/ends_with.rs

```diff
@@ -42,17 +41,7 @@ impl Default for EndsWithFunc {
 impl EndsWithFunc {
     pub fn new() -> Self {
         Self {
-            signature: Signature::one_of(
-                vec![
-                    // Planner attempts coercion to the target type starting with the most preferred candidate.
-                    // For example, given input `(Utf8View, Utf8)`, it first tries coercing to `(Utf8View, Utf8View)`.
-                    // If that fails, it proceeds to `(Utf8, Utf8)`.
-                    Exact(vec![DataType::Utf8View, DataType::Utf8View]),
-                    Exact(vec![DataType::Utf8, DataType::Utf8]),
-                    Exact(vec![DataType::LargeUtf8, DataType::LargeUtf8]),
-                ],
-                Volatility::Immutable,
-            ),
+            signature: Signature::string(2, Volatility::Immutable),
```

Comment:
I think it's actually incorrect. My understanding of `arrow::compute::kernels::comparison::ends_with` is that it actually supports exactly only the cases where left type == right type.

Reply:
As you pointed out above, doesn't Signature::string coerce to a single common type?

Reply:
that was written before i realized i read the code wrong. resolving